### PR TITLE
Validate overlapping periods

### PR DESCRIPTION
**Pull request by BacLuc**
Rather minimalistic implementation.
Screenshot 2022-11-26 at 16-27-10 eCamp3
Screenshot 2022-11-26 at 16-28-20 eCamp3
Issue: #2553

### CODE REVIEW

**Review by carlobeltrame — APPROVED** [review]
Thanks!

File: api/tests/Api/Camps/CreateCampTest.php [hunk 1]
```diff
@@ -111,6 +111,50 @@ public function testCreateCampValidatesInvalidPeriods() {
         ]);
     }
 
+    public function testCreateCampValidatesOverlappingPeriods() {
+        static::createClientWithCredentials()->request('POST', '/camps', ['json' => $this->getExampleWritePayload([
+            'periods' => [
+                [
+                    'description' => 'Aufbau',
+                    'start' => '2022-01-07', // start date after end date is invalid
```
Comment (usu) [inline api/tests/Api/Camps/CreateCampTest.php]:
```suggestion
                    'start' => '2022-01-07',
```

**Review by usu — APPROVED** [review]
Thanks! 

Only small ciomment on the naming of the validator. I leave it to you if you want to change or keep the existing one.

When I see `AssertNoOverlappingPeriods`, then I read: "Assert that none of the period is overlapping"
I would expect a validator with such a name for example on the `periods` property in the `Camp` entity.

What the validator actually is doing: "Assert that the given value (DateTime) is not overlapping with any other period"

It's a small but fine difference. A name that would more accurately represent this logic could for example be `AssertNotOverlappingWithOtherPeriods`

File: api/tests/Api/Camps/CreateCampTest.php (hunk 1 above)
Comment (BacLuc) [reply api/tests/Api/Camps/CreateCampTest.php]: done

**Review by usu — APPROVED** [review]

**Pull request merged**